common/djangoapps/external_auth/views.py

```diff
@@ -478,13 +478,11 @@ def cas_login(request, next_page=None, required=False):
 
     if request.user.is_authenticated():
         user = request.user
-        UserProfile.objects.get_or_create(
-            user=user,
-            defaults={'name': user.username}
-        )
-        Subscriber.objects.get_or_create(
-            user=user
-        )
+        profile, _ = UserProfile.objects.get_or_create(user=user)
+        profile.name = user.get_full_name()
+        profile.save()
```

Comment:
why not keep using `get_or_create(user=user, defaults={'name': user.get_full_name()})`? it seems like 1 less database write with no downsides